### PR TITLE
Caffe multigpu

### PR DESCRIPTION
**Pull request by sat3ll**
Allows users to specify other or more gpus.

### CODE REVIEW

File: bin/video2x.py [hunk 1]
```diff
@@ -260,6 +261,35 @@ def absolutify_paths(config):
...
+if args.devices and not args.driver == 'waifu2x_caffe':
+    Avalon.error('Using multiple devices is only available for waifu2x_caffe')
+    exit(1)
+
+# check if gpu ids supplied are actually valid
+if args.devices:
+    # filter duplicate gpu ids
+    args.devices = list(set(args.devices))
+
+    available_gpulist = []
```
Comment (k4yt3x) [inline bin/video2x.py]: This can be shorter

**Comment by 28598519a** [comment]
auto select available GPU Devices

`args.devices = GPUtil.getAvailable(limit = 8) `

**Comment by tamodolo** [comment]
This is actualy a great feature. Please consider when implementing this that user can have diferent GPU brands (like 1 nVidia and 1 AMD). So alowing that for drivers that have tech in common will be usefull.

**Comment by nihui** [comment]
I added multi gpu support for waifu2x-ncnn-vulkan

```
./waifu2x-ncnn-vulkan -g 0,1
```

Then it will use both gpu-0 and gpu-1 for upscaling tasks
Should work on any device supporting vulkan api

**Comment by k4yt3x** [comment]
I think this can be closed. With *-ncnn-vulkan adding multi-GPU support, I don't see the value of investing time in adding multi-GPU support for other drivers. It's already a luxury for me to spend time working on side-projects. I'll have to leave multi-GPU waifu2x-caffe for later if it ever happens.

**Comment by lovesegfault** [comment]
@k4yt3x Came across this today, how is this expected to work? if I pass `-g 0,1` I end up with video2x complaining that the value is not an int. Same happens if I use a config file.

**Comment by k4yt3x** [comment]
@lovesegfault ah that's a good point, I'll have to update the parser for it to work properly.

**Comment by lovesegfault** [comment]
> @lovesegfault ah that's a good point, I'll have to update the parser for it to work properly.

I'm hacking on this locally, it's solvable by just making the `-t` and `-g` arguments strings instead of ints.

I'll report back if I get it working :)

**Comment by k4yt3x** [comment]
The GUI also needs to be updated. Care to open an issue for this?

**Comment by lovesegfault** [comment]
> The GUI also needs to be updated. Care to open an issue for this?

Done! https://github.com/k4yt3x/video2x/issues/436

**Pull request closed without merging**